src/foam/demos/examples/services.fbe

```diff
@@ -0,0 +1,435 @@
+## Services
+
+FOAM services provide a clean way to implement distributed functionality with client-server communication. Services automatically generate client stubs, server skeletons, and handle network marshalling.
+
+## Service Architecture
+
+The typical FOAM service consists of:
+
+1. **Interface Definition** - Defines the service contract
+2. **Server Implementation** - Contains business logic  
+3. **Service Registration** - Configures deployment in services.jrl
+4. **Client Usage** - Import and use via context injection
+
+##  Basic Service Interface
+
+Define a service interface using foam.INTERFACE() with key properties:
+- `skeleton: true` - Auto-generates server skeleton
+- `client: true` - Auto-generates client stub  
+- `async: true` - Required for methods returning values
+--
+foam.INTERFACE({
+  package: 'foam.demo.service',
+  name: 'TaskService',
+  
+  skeleton: true,
+  client: true,
+  
+  methods: [
+    {
+      name: 'createTask',
+      async: true,
+      type: 'String',  
+      args: 'String title, String description'
+    },
+    {
+      name: 'getTask',
+      async: true,
+      type: 'Object',
+      args: 'String taskId'
+    },
+    {
+      name: 'deleteTask',
+      async: true, 
+      type: 'boolean',
+      args: 'String taskId'
+    }
+  ]
+});
+
+log('TaskService interface defined');
+
+##  Server Implementation
+
+Implement the service interface with business logic:
+--
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'SimpleTaskService',
```

Comment:
Typically, the 'server' or implementation is server side which means Java (currently), so this example needs **javaCode**: blocks rather than **code:** blocks, and the property needs a **class**:  definition. 
If you are planning on running this from a **nodejs** server then this would work, but that would not be a very useful server. 

Ok, read through the entire PR and see you are only working from the client, which is not a real world scenario, so I don't see this PR as a guide to 'best practice'.

Reply:
Refactored and pushed. sorry for that